File: code/1.两数之和.java
/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] indexs = new int[2];
        HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++) {
            if (hash.containsKey(nums[i])) {
                indexs[0] = i;
                indexs[1] = hash.get(nums[i]);
                return indexs;
            }
            //将数据存入  value为下标
            hash.put(target-nums[i], i);
        }
        return indexs;
    }
}
// @lc code=end

File: code/1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
package code

func twoSum(nums []int, target int) []int {
	set := map[int]int{}
	for i, v := range nums {
		if _, ok := set[target-v]; ok {
			return []int{i, set[target-v]}
		}
		set[v] = i
	}
	return []int{}
}

// @lc code=end
